### ISSUE TITLE
Create/edit Smart Search for View

### ISSUE THREAD

**Issue by richardolsson** [zetkin/app.zetkin.org#474]
## Description
A user should be able to configure a Smart Search query in a view, so that the view is dynamically populated with the people matching that Smart Search query. When the result of the query changes, so does the content of the view.

This is different from a "static" view, where people are added one by one to the view (or from another view, as described in #476), and thus, the user needs to be able to pick whether the view should be static or dynamic.

## Relevant Job Stories
* When I'm brainstorming people to add to my view as a shortlist, I want to be able to define a Smart Search query and see the resulting data set in my view, so that I can add the ones I want to the view from there
* When I'm using a Smart Search query to populate my view, I want to be able to save it in the view so that every time I come back the view contains the people currently matching that query

## Prerequisites
This task requires changes to the Views API. See updates in comments below, or ask if unsure.

## Requirements
- [x] On the single-view page, if a view is empty, it should ask the user to decide whether to statically add specific people or define a Smart Search
- [x] If user picks static, render the empty table and the line where user can add records (#475)
- [x] If user picks Smart Search, open the Smart Search edit modal, and save the Smart Search to the view

## Possible implementations
Use the existing Smart Search modal to define Smart Search. Attach the query to the view using `PATCH /orgs/{org_id}/people/views/{view_id}` (pending changes to API).

## Design specifications
![image](https://user-images.githubusercontent.com/550212/143042359-bf21c383-920d-4e8b-a285-e29cfcad4dcf.png)

## Open questions
No outstanding questions at this time.

**Comment by richardolsson** [comment]
I've taken this on, and I'd like to add a slight twist which is that it should be possible to add the first person to the view directly from this screen, instead of first picking "static" and then adding. I could do this by adding a person search/select widget to the box that is not about Smart Search (but should probably no longer be called "Create empty view" if I do that).

**Comment by richardolsson** [comment]
I've made changes to the Platform API which adds a `content_query` attribute to Views (and also a `{view_id}/content_query` endpoint where it can be edited). 

There are actually four states that need to be handled differently in the UI:

### 1. "Unknown": No `content_query`, has no rows
This is the initial state when a new view has been created, before either a `content_query` or a first row (person) has been added. This is the state where the original wireframe makes sense. Requirements in this state are:

- [ ] User should be able to add a Smart Search query
- [ ] User should be able to add a first row

### 2. "Dynamic empty": Has `content_query`, has no rows
This is a state where the user has configured a `content_query`, but that query does not currently return any matches, so the view is empty. In this case, the user should be made aware that that's what has happened. Requirements:

- [ ] User should be notified that the view is empty because the query did not return any rows
- [ ] User should be able to edit the `content_query`
- [ ] User should be able to convert the view to static (discard the `content_query`)

### 3. "Dynamic content": Has `content_query`, has rows
This is a state where the user has configured a `content_query`, and that query presently results in at least one match, so the view will have at least one row. It's identical in most ways to 4 (below). The requirement is this

- [ ] User should be able to edit the `content_query`
- [ ] User should be able to convert the view to static (discard the `content_query`)

This could be solved via options in the ellipsis menu.

### 4. "Static content": No `content_query`, has rows
This is a state where the user has added at least one row. It's identical in most ways to 3 (above), the only difference in requirements being that:

- [ ] User should be able to convert to dynamic (add a `content_query`)

This could be solved via an option in the ellipsis menu.

**Comment by richardolsson** [comment]
I've created a little proof of concept which feels very good to use, although not very beautiful yet.

![image](https://user-images.githubusercontent.com/550212/146667916-eec04752-5695-44c7-b534-3b1277720ac6.png)

I realized a bunch of stuff while hacking it together, so I will start from scratch based on my findings. Here's my battle plan:

1. Create a test for the Task Assignees page (because it uses the Smart Search dialog)
2. Refactor Smart Search dialog to return edited query in callback instead of mutating the API
3. Improve test for creating new campaign to include assigning a manager
4. Break out the campaign manager widget from `CampaignDetailsForm` to new `PersonSelect` widget with tests
5. Implement the two missing view states 1 and 2 (from previous comment), using the Smart Search dialog and `PersonSelect` widget
6. Create ellipsis menu and add options to convert between static/dynamic and to edit Smart Search query
